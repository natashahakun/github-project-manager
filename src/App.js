import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { LoadingIndicator, Login, Notification } from './components';
import Dashboard from './views/Dashboard/Dashboard';
import { getUser } from './actions/user.actions';
import { apiKeyService } from './services';

class App extends Component {
    state = { checkedExisting: false }

    async componentDidMount() {
        const existingApiKey = apiKeyService.get();
        existingApiKey && await this.props.getUser(existingApiKey);
        this.setState({ checkedExisting: true });
    }

    render() {
        const { errorMessage, isLoading, user } = this.props;
        const { checkedExisting } = this.state;

        return (
            <div>
                { errorMessage && <Notification>{ errorMessage }</Notification>}
                { isLoading && <LoadingIndicator /> }

                { user && <Dashboard /> }
                { !user && checkedExisting && <Login /> }
            </div>
        )
    }
}

    
App.defaultProps = {
    errorMessage: '',
    user: ''
};

App.propTypes = {
    errorMessage: PropTypes.string,
    getUser: PropTypes.func,
    isLoading: PropTypes.bool.isRequired,
    user: PropTypes.string
};

const mapStateToProps = ({ ui, user }) => ({
    errorMessage: ui.error,
    isLoading: ui.loading,
    user: user.githubUsername
});

const mapDispatchToProps = dispatch => bindActionCreators({
    getUser
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App);
