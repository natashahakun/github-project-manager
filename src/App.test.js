import React from 'react';
import renderer from 'react-test-renderer';
import { mockStore } from './testConfig';
import { Provider } from 'react-redux';
import App from './App';

let store;

beforeEach(() => {
    store = mockStore({ 
        ui: {
            loading: false
        }, 
        user: {},
        getState: jest.fn(() => ({})),
        dispatch: jest.fn()
    });
});

describe('App', () => {
    it('renders correctly', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
            <App />
            </Provider>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
