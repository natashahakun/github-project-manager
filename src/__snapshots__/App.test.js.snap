// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  
  
  <div
    className="login"
  >
    <div
      className="login__body"
    >
      <h1
        className="heading"
      >
        Login
      </h1>
      <p
        className="login__text"
      >
        Enter GitHub token to access your GitHub Project Manager
      </p>
      <form
        className="login__form"
        onSubmit={[Function]}
      >
        <div
          className="input-wrapper"
        >
          <label
            className="input-wrapper__label"
            htmlFor="apiKey"
          >
            GitHub Token
          </label>
          <input
            className="input-wrapper__input"
            id="apiKey"
            name="apiKey"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="login__action"
        >
          <button
            className="button button--primary"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
      <div
        className="link-wrapper"
      >
        <a
          href="https://help.github.com/en/enterprise/2.17/user/articles/creating-a-personal-access-token-for-the-command-line"
          rel="noopener noreferrer"
          target="_blank"
        >
          Find out how to obtain a GitHub personal access token
        </a>
      </div>
    </div>
  </div>
</div>
`;
